XXX-icu-module.md

```diff
@@ -0,0 +1,137 @@
+| Title  | ICU Module                  |
+|--------|-----------------------------|
+| Author | @jasnell                    |
+| Status | REJECTED                    |
+| Date   | 2016-04-20                  |
+
+## Description
+
+The ICU4C library that we use for internationalization contains a significant
+array of additional functionality not currently exposed by the EcmaScript 402
+standard. Some of this additional functionality would be useful to expose via
+a new `'icu'` or (`'uncode'`) module.
+
+## Interface
+
+Initially, the `'icu'` module would provide methods for the following:
+
+1. Character encoding detection. ICU includes code that is able to look at a
+   stream of bytes and apply heuristics to detect the character encoding in
+   use. This is not always an exact match but it does a reasonably good job.
+   We can tune this detection to only look for the character encodings we
+   support in Core (ascii, iso8859-1, utf8 and utf16-le). Two specific APIs
```

Comment:
As far as I know, yes, but it might require the full-icu data set
currently. Our small-icu filters out much of the non-unicode character
encoding support. I can look into it tho. /cc @srl295
On Apr 27, 2016 2:58 PM, "Trevor Norris" <notifications@github.com> wrote:

> In XXX-icu-module.md
> <https://github.com/nodejs/node-eps/pull/15#discussion_r61344015>:
>
> > +## Description
> > +
> > +The ICU4C library that we use for internationalization contains a significant
> > +array of additional functionality not currently exposed by the EcmaScript 402
> > +standard. Some of this additional functionality would be useful to expose via
> > +a new `'icu'` or (`'uncode'`) module.
> > +
> > +## Interface
> > +
> > +Initially, the `'icu'` module would provide methods for the following:
> > +
> > +1. Character encoding detection. ICU includes code that is able to look at a
> > +   stream of bytes and apply heuristics to detect the character encoding in
> > +   use. This is not always an exact match but it does a reasonably good job.
> > +   We can tune this detection to only look for the character encodings we
> > +   support in Core (ascii, iso8859-1, utf8 and utf16-le). Two specific APIs
>
> does icu support win1252? that would be helpful for parsing data from
> browsers. since the WHATWG has listed that iso8859-1 be a sym for win1252,
> so it'd be impossible to convert using 'binary' encoding.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node-eps/pull/15/files/3fad3f260a0533ef0fead0d640ac2c3cc78d2d0a#r61344015>
>